src/Language/C/Parser/Lexer.x

```diff
@@ -302,88 +302,88 @@ label __label__
+idkwtok ('_' : '_' : 'i' : 'n' : 'l' : 'i' : 'n' : 'e' : '_' : '_' : []) pos = tok 10 CTokInline pos
+idkwtok ('i' : 'n' : 't' : []) pos = tok 3 CTokInt pos
+idkwtok ('_' : '_' : 'i' : 'n' : 't' : '1' : '2' : '8' : []) pos = tok 8 CTokInt128 pos
+idkwtok ('_' : '_' : 'l' : 'a' : 'b' : 'e' : 'l' : '_' : '_' : []) pos = tok 9 CTokLabel pos
+idkwtok ('l' : 'o' : 'n' : 'g' : []) pos = tok 4 CTokLong pos
+idkwtok ('_' : '_' : 'n' : 'o' : 'n' : 'n' : 'u' : 'l' : 'l' : []) pos = tok 9 CTokNonnull pos
+idkwtok ('_' : '_' : 'n' : 'u' : 'l' : 'l' : 'a' : 'b' : 'l' : 'e' : []) pos = tok 10 CTokNullable pos
+idkwtok ('_' : '_' : 'r' : 'e' : 'a' : 'l' : []) pos = tok 6 (CTokGnuC GnuCComplexReal) pos
+idkwtok ('_' : '_' : 'r' : 'e' : 'a' : 'l' : '_' : '_' : []) pos = tok 8 (CTokGnuC GnuCComplexReal) pos
+idkwtok ('r' : 'e' : 'g' : 'i' : 's' : 't' : 'e' : 'r' : []) pos = tok 8 CTokRegister pos
+idkwtok ('_' : '_' : 'r' : 'e' : 's' : 't' : 'r' : 'i' : 'c' : 't' : []) pos = tok 10 CTokRestrict pos
+idkwtok ('r' : 'e' : 's' : 't' : 'r' : 'i' : 'c' : 't' : []) pos = tok 8 CTokRestrict pos
+idkwtok ('_' : '_' : 'r' : 'e' : 's' : 't' : 'r' : 'i' : 'c' : 't' : '_' : '_' : []) pos = tok 12 CTokRestrict pos
+idkwtok ('r' : 'e' : 't' : 'u' : 'r' : 'n' : []) pos = tok 6 CTokReturn pos
+idkwtok ('s' : 'h' : 'o' : 'r' : 't' : []) pos = tok 5 CTokShort pos
+idkwtok ('_' : '_' : 's' : 'i' : 'g' : 'n' : 'e' : 'd' : []) pos = tok 8 CTokSigned pos
+idkwtok ('s' : 'i' : 'g' : 'n' : 'e' : 'd' : []) pos = tok 6 CTokSigned pos
+idkwtok ('_' : '_' : 's' : 'i' : 'g' : 'n' : 'e' : 'd' : '_' : '_' : []) pos = tok 10 CTokSigned pos
+idkwtok ('s' : 'i' : 'z' : 'e' : 'o' : 'f' : []) pos = tok 6 CTokSizeof pos
+idkwtok ('s' : 't' : 'a' : 't' : 'i' : 'c' : []) pos = tok 6 CTokStatic pos
+idkwtok ('s' : 't' : 'r' : 'u' : 'c' : 't' : []) pos = tok 6 CTokStruct pos
+idkwtok ('s' : 'w' : 'i' : 't' : 'c' : 'h' : []) pos = tok 6 CTokSwitch pos
+idkwtok ('_' : '_' : 't' : 'h' : 'r' : 'e' : 'a' : 'd' : []) pos = tok 8 CTokThread pos
+idkwtok ('t' : 'y' : 'p' : 'e' : 'd' : 'e' : 'f' : []) pos = tok 7 CTokTypedef pos
+idkwtok ('_' : '_' : 't' : 'y' : 'p' : 'e' : 'o' : 'f' : []) pos = tok 8 CTokTypeof pos
+idkwtok ('t' : 'y' : 'p' : 'e' : 'o' : 'f' : []) pos = tok 6 CTokTypeof pos
+idkwtok ('_' : '_' : 't' : 'y' : 'p' : 'e' : 'o' : 'f' : '_' : '_' : []) pos = tok 10 CTokTypeof pos
+idkwtok ('u' : 'n' : 'i' : 'o' : 'n' : []) pos = tok 5 CTokUnion pos
+idkwtok ('u' : 'n' : 's' : 'i' : 'g' : 'n' : 'e' : 'd' : []) pos = tok 8 CTokUnsigned pos
+idkwtok ('v' : 'o' : 'i' : 'd' : []) pos = tok 4 CTokVoid pos
+idkwtok ('_' : '_' : 'v' : 'o' : 'l' : 'a' : 't' : 'i' : 'l' : 'e' : []) pos = tok 10 CTokVolatile pos
+idkwtok ('v' : 'o' : 'l' : 'a' : 't' : 'i' : 'l' : 'e' : []) pos = tok 8 CTokVolatile pos
+idkwtok ('_' : '_' : 'v' : 'o' : 'l' : 'a' : 't' : 'i' : 'l' : 'e' : '_' : '_' : []) pos = tok 12 CTokVolatile pos
+idkwtok ('w' : 'h' : 'i' : 'l' : 'e' : []) pos = tok 5 CTokWhile pos
```

Comment:
Interesting, the Core did change after I eta-expanded `idkwtok` but only by floating some let bindings from the CTokIdent/ CTokTyIdent case, below.   I don't see anything that would suggest fewer allocations.